src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCaseTrait.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use PHPUnit\Framework\TestCase;
+
+// Auto-adapt to PHPUnit 8 who added a `void` return-type to the tearDown method
+
+if (method_exists(\ReflectionMethod::class, 'hasReturnType') && (new \ReflectionMethod(TestCase::class, 'tearDown'))->hasReturnType()) {
+eval('
+    /**
+     * @internal
+     */
+    trait KernelTestCaseTrait
+    {
+        protected function tearDown(): void
+        {
+            static::ensureKernelShutdown();
+        }
+    }
+');
+} else {
+    /**
+     * @internal
+     */
+    trait KernelTestCaseTrait
```

Comment:
I suggest to use a more specific trait name like `KernelShutdownOnTestCaseTearDownTrait` because we might want to use the current name for a trait that is currently in KernelTestCase class. I think having this as a trait makes more sense because it makes it independent from phpunit and more flexible. Also IMO we should not include the automatic kernel shutdown in the future because it makes funtional tests really slow.